File: lismove-android/app/src/main/java/it/lismove/app/android/session/apiService/OfflineDataRequest.kt
package it.lismove.app.android.session.apiService

import net.nextome.lismove_sdk.models.LisMoveSensorHistoryElement

data class OfflineDataRequest(
    val distance: Float?,
    val endRevs: Int?,
    val endTime: Long?,
    val sensor: String?,
    val startRevs: Int?,
    val startTime: Long?,
    val user: String?, )

fun LisMoveSensorHistoryElement.asOfflineDataRequest(userId: String?, sensorName: String?) =
    OfflineDataRequest(distance, stopLap, stopUtc, sensorName, startLap, startUtc, userId)

File: lismove-iOS/LisMove/Modules/Shared/Model/OfflineDataRequest.swift
//
//  OfflineDataRequest.swift
//  LisMove
//
//

import Foundation
public struct OfflineDataRequest: Codable{
    public let distance: Float?
    public let endRevs: Int?
    public let endTime: Int?
    public let sensor: String?
    public let startRevs: Int?
    public let startTime: Int?
    public let user: String?
}

